collections/_article/introducing-reverse-z.md

```diff
@@ -0,0 +1,135 @@
+---
+title: "Introducing Reverse Z (AKA I'm sorry for breaking your shader)"
+excerpt: "We are breaking compatibility for some custom shaders. Here is why."
+categories: ["news"]
+author: Clay John
+image: /storage/blog/covers/reverse-z.webp
+date: 2024-04-19 15:00:00
+---
+
+After extensive discussion, we have decided to implement the reverse z depth buffer technique in Godot 4.3. This is an exciting change as it brings a massive improvement to depth buffer precision at no performance or memory cost. This technique is used everywhere in 3D games these days. In practical terms, it significantly reduces the chances of running into z-fighting and other depth buffer precision artifacts. NVIDIA has an [excellent article](https://developer.nvidia.com/content/depth-precision-visualized) explaining the theory and benefits behind using reverse z, please read it for more technical info.
+
+I am writing this post because, unfortunately, implementing reverse z naturally breaks compatibility for some shaders. We try to avoid compatibility breakage as much as possible, but in some cases it is unavoidable, or the benefits of doing so far outweigh the cost. The rendering team felt in this case that the benefits sufficiently outweighed the costs.
+
+We are certain that the vast majority of users will not run into any compatibility breakage. For most people this change is all good with no downside. 
+
+You may need to tweak your shaders if you use a custom spatial Shader that:
```

Comment:
```suggestion
You may need to tweak your shaders if you use a custom spatial shader that:
```
Unless the capital S is significant..?

Reply:
It is, but it doesn't have to be. I usually capitalize words that are directly referring to a Godot class of the same name as this is the style we use in the docs. So "Shader" would refer to Godot's Shader class, while "shader" would refer to the concept of shaders in general. Here I capitalize because I am talking about a Godot Shader and not the concept of shaders in general. 

I don't have a strong opinion about this. Like I said, I was just copying the docs style here, if it looks awkward and out of place, I am happy to change it.

Reply:
Since this is the only time I refer to the shader class directly, it does seem a little out of place. I have made it lowercase now